components/selection/test/selection.vdiff.js

```diff
@@ -0,0 +1,313 @@
+import '../selection-action.js';
+import '../selection-action-dropdown.js';
+import '../selection-controls.js';
+import '../selection-input.js';
+import '../selection-select-all.js';
+import '../selection-select-all-pages.js';
+import '../selection-summary.js';
+import './selection-component.js';
+import { clickElem, expect, fixture, focusElem, html, oneEvent, sendKeysElem } from '@brightspace-ui/testing';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { nothing } from 'lit';
+
+const viewport = { width: 476 };
+
+function createInputList(numInputs, selected) {
+	return html`
+		<ul style="padding: 0;">
+			${Array.from(Array(numInputs).keys()).map((key) => html`
+				<li style="list-style-type: none;">
+					<d2l-selection-input
+						style="margin-right: 10px;"
+						label="${`item ${key + 1}`}"
+						key="${`key${key + 1}`}"
+						?selected="${selected && selected[key]}">
+					</d2l-selection-input>${`Item ${key + 1}`}
+				</li>
+			`)}
+		</ul>
+	`;
+}
+
+describe('selection-components', () => {
+	[{
+		category: 'action',
+		tests: [
+			{ name: 'text', template: html`<d2l-selection-action text="action"></d2l-selection-action>` },
+			{ name: 'text-icon', template: html`<d2l-selection-action text="action" icon="tier1:gear"></d2l-selection-action>` },
+			{ name: 'disabled', template: html`<d2l-selection-action text="action" disabled></d2l-selection-action>`, waitFor: elem => oneEvent(elem, 'd2l-tooltip-show') }
+		],
+		requiresSelectionTemplate: html`<d2l-selection-action text="action" requires-selection></d2l-selection-action>`
+	},
+	{
+		category: 'dropdown',
+		tests: [
+			{ name: 'text', template: html`<d2l-selection-action-dropdown text="action"></d2l-selection-action-dropdown>` },
+			{ name: 'disabled', template: html`<d2l-selection-action-dropdown text="action" disabled></d2l-selection-action-dropdown>`, waitFor: elem => oneEvent(elem, 'd2l-tooltip-show') }
+		],
+		requiresSelectionTemplate: html`<d2l-selection-action-dropdown text="action" requires-selection></d2l-selection-action-dropdown>`
+	}].forEach(({ category, tests, requiresSelectionTemplate }) => {
+		describe(category, () => {
+			tests.forEach(({ name, template, waitFor }) => {
+				it(name, async() => {
+					const elem = await fixture(template);
+					await expect(elem).to.be.golden();
+				});
+
+				it(`${name}-focus`, async() => {
+					const elem = await fixture(template);
+					await focusElem(elem);
+					if (waitFor) await waitFor(elem);
+					await expect(elem).to.be.golden();
+				});
+			});
+
+			[
+				{ name: 'requires-selection-none', selectionInfo: { state: 'none', keys: [] } },
+				{ name: 'requires-selection-some', selectionInfo: { state: 'some', keys: [] } },
+				{ name: 'requires-selection-all', selectionInfo: { state: 'all', keys: [] } }
+			].forEach(({ name, selectionInfo }) => {
+				it(name, async() => {
+					const elem = await fixture(requiresSelectionTemplate);
+					elem.selectionInfo = selectionInfo;
```

Comment:
I guess it doesn't need it, but surprised this doesn't require a `await elem.updateComplete` after setting the info.

Reply:
Yeah, it _might_ be because calculating the rect forces a paint with the `getBoundingClientRect` calls? Although the old visual diffs also didn't need it